blockchain/state/statedb.go

```diff
@@ -402,7 +402,7 @@ func (s *StateDB) GetVmVersion(addr common.Address) (params.VmVersion, bool) {
 	stateObject := s.getStateObject(addr)
 	if stateObject != nil {
 		pa := account.GetProgramAccount(stateObject.account)
-		if pa != nil {
+		if pa != nil && pa.Type() == account.SmartContractAccountType {
```

Comment:
Why this condition? Does it break any test?
Both EOAwithCode and SCA can have a VmVersion of its own. Also VmVersion1 should be the correct number for EOAwithCode (see https://github.com/kaiachain/kaia/blob/854b297dbef52bbc11ba60413a47110e157ef525/params/protocol_params.go#L282)

Reply:
How do I correctly create EOA object? VmVersion0 is set when EOA is created by statedb.SetState or statedb.CreateEOA.

Reply:
https://github.com/kaiachain/kaia/blob/932de969fd197fbb5c1cf594bc8ac30ce6a0a455/blockchain/vm/evm.go#L631-L655
When EOA tries to call a precompiled contract, it can't get latest precompiled contracts.

Reply:
Oh the EOA not having the correct VmVersion is the root cause then.

- For SCAs, it is enforced by `statedb.CreateSmartContractAccount` where `CodeInfo = NewCodeInfoWithRules` is set.
- For EOAs, it should be enforced in SetCodeToEOA (because CodeInfo was not set when EOA is created)

I think I should modify KIP-228 to say
> Set the code of authority to be `0xef0100 || address` and set the codeinfo of the authority to be `0x11 (EVM & VmVersion1)`

@Mdaiki0730 @shiki-tak